### PR TITLE
Add self observability metrics for otel handler

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

**Comment by kezhenxu94** [comment]
image

Why is the latency percentile so huge? Did I do anything wrong?

**Comment by wu-sheng** [comment]
Buckets of histogram should be as close as latency layout as possible. The the percentiles from buckets make the sense.

**Comment by kezhenxu94** [comment]
Looks like the latency is too huge indeed.

Please check the demo http://34.92.247.5/dashboard/SO11Y_OAP/ServiceInstance/ZGVtby1vYXAuc2t5d2Fsa2luZy1zaG93Y2FzZQ==.1/ZGVtby1vYXAuc2t5d2Fsa2luZy1zaG93Y2FzZQ==.1_ZGVtby1vYXAtNzY1NjdjNjhjZi1xZnE3ZA==/Self-Observability-OAP-Instance

**Comment by wu-sheng** [comment]
Which buckets are you using? If last is 10ms, then over 10ms means max.int.

**Comment by wu-sheng** [comment]
That percentile of histogram is not OAL percentile functions. Skywalking percentile in oal is auto adoptive, but otlp/Prometheus is predefined.

File: oap-server/server-receiver-plugin/otel-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/otel/otlp/OpenTelemetryLogHandler.java [hunk 1]
```diff
@@ -69,30 +93,32 @@ public void active() throws ModuleStartException {
...
-                .getScopeLogsList()
-                .stream()
-                .flatMap(it -> it.getLogRecordsList().stream())
-                .forEach(
-                    logRecord ->
-                        doAnalysisQuietly(service, layer, serviceInstance, logRecord));
-            responseObserver.onNext(ExportLogsServiceResponse.getDefaultInstance());
-            responseObserver.onCompleted();
-        });
+        getReceivedCounter().inc();
```
[the remark this answers is not in the document]
Comment (kezhenxu94) [reply oap-server/server-receiver-plugin/otel-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/otel/otlp/OpenTelemetryLogHandler.java:96]: > counter of bulk could be very misleading to low payload, which actually is not that low.

It’s not about bulk or not bulk, what I want to measure here is the requests received, not the metrics count inside the request. If we want to further load balance the traffic in the future in Satellite, I think measuring in the dimension of requests makes more sense than in the dimension of metrics.

File: oap-server/server-receiver-plugin/otel-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/otel/otlp/OpenTelemetryMetricRequestProcessor.java [hunk 2]
```diff
@@ -73,39 +79,60 @@ public class OpenTelemetryMetricRequestProcessor implements Service {
...
+                            .flatMap(scopeMetrics -> scopeMetrics
+                                .getMetricsList().stream()
+                                .flatMap(metric -> adaptMetrics(nodeLabels, metric))
+                                .map(Function1.liftTry(Function.identity()))
+                                .flatMap(tryIt -> MetricConvert.log(
+                                    tryIt,
+                                    "Convert OTEL metric to prometheus metric"
+                                )))));
+            });
+        }
```
Comment (kezhenxu94) [inline oap-server/server-receiver-plugin/otel-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/otel/otlp/OpenTelemetryMetricRequestProcessor.java:135]: @wu-sheng also, due to the logic `convert.toMeter`, it processes all the metrics in a batch, we have no way to measure the latency metric by metric

Comment (wu-sheng) [reply oap-server/server-receiver-plugin/otel-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/otel/otlp/OpenTelemetryMetricRequestProcessor.java:135]: OK. It is fine. But still, counter is not necessary, as timer/histogram includes a timer.

**Comment by wu-sheng** [comment]
Could you share the latest dashboard of so11y? I think new are added into inbound traffic.

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/so11y_oap/so11y-instance.json [hunk 3]
```diff
@@ -467,6 +467,148 @@
...
+                  "graph": {
+                    "type": "Line",
+                    "step": false,
+                    "smooth": false,
+                    "showSymbol": true,
+                    "showXAxis": true,
+                    "showYAxis": true
+                  },
+                  "widget": {
+                    "title": "OpenTelemetry Metrics (Request / Second)"
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/so11y_oap/so11y-instance.json:490]:
```suggestion
                    "title": "OpenTelemetry Metrics (Requests / Second)"
```

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/so11y_oap/so11y-instance.json [hunk 4]
```diff
@@ -467,6 +467,148 @@
...
+                  "graph": {
+                    "type": "Line",
+                    "step": false,
+                    "smooth": false,
+                    "showSymbol": true,
+                    "showXAxis": true,
+                    "showYAxis": true
+                  },
+                  "widget": {
+                    "title": "OpenTelemetry Logs (Request / Second)"
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/so11y_oap/so11y-instance.json:517]:
```suggestion
                    "title": "OpenTelemetry Logs (Requests / Second)"
```

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/so11y_oap/so11y-instance.json [hunk 5]
```diff
@@ -467,6 +467,148 @@
...
+                  "graph": {
+                    "type": "Line",
+                    "step": false,
+                    "smooth": false,
+                    "showSymbol": true,
+                    "showXAxis": true,
+                    "showYAxis": true
+                  },
+                  "widget": {
+                    "title": "OpenTelemetry Spans (Request / Second)"
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/so11y_oap/so11y-instance.json:539]:
```suggestion
                    "title": "OpenTelemetry Spans (Requests / Second)"
```

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/so11y_oap/so11y-instance.json [hunk 6]
```diff
@@ -467,6 +467,148 @@
...
+                  },
+                  "metricConfig": [
+                    {
+                      "unit": "Request / Second"
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/so11y_oap/so11y-instance.json:543]:
```suggestion
                      "unit": "Requests / Second"
```

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/so11y_oap/so11y-instance.json [hunk 7]
```diff
@@ -467,6 +467,148 @@
...
+                },
+                {
+                  "x": 18,
+                  "y": 13,
+                  "w": 6,
+                  "h": 13,
+                  "i": "22",
+                  "type": "Widget",
+                  "expressions": [
+                    "relabels(meter_oap_otel_metrics_latency_percentile{p='50,75,90,95,99'},p='50,75,90,95,99',percentile='50,75,90,95,99')"
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/so11y_oap/so11y-instance.json:555]: Please add the unit. I think it's `ms`?

**Comment by kezhenxu94** [comment]
> Could you share the latest dashboard of so11y? I think new are added into inbound traffic.

Please check the latest demo env, it was updated before those suggested comments.

**Comment by kezhenxu94** [comment]
http://34.92.247.5/dashboard/SO11Y_OAP/ServiceInstance/ZGVtby1vYXAuc2t5d2Fsa2luZy1zaG93Y2FzZQ==.1/ZGVtby1vYXAuc2t5d2Fsa2luZy1zaG93Y2FzZQ==.1_ZGVtby1vYXAtNzlkZjk0OWY1Yy1yZHN4Zg==/Self-Observability-OAP-Instance

**Review by wu-sheng — APPROVED** [review]

**Review by wankai123 — APPROVED** [review]

**Comment by wu-sheng** [comment]
Let's bring the latest to the showcase. We can check and see later. I hope the recent repeatable crashing will be fixed.

**Pull request merged**